### PR TITLE
DEV-29 Add networking model

### PR DESCRIPTION
**Pull request by lukaszcettler**
Added service networking connected with Login View, user can log in and use the app. Implemented Add donation and disqualification views, added profile view with medals, added navigation.

### CODE REVIEW

File: BloodDonor/Modules/Donation/DetailDonation.swift
Comment (adjurk) [inline BloodDonor/Modules/Donation/DetailDonation.swift:1]: We should follow the convention of 'SomethingDetail', similarly to my already prepared `BloodCenterDetail`

File: BloodDonor/Modules/DonationList/DonationListView.swift [hunk 1]
```diff
@@ -10,10 +10,13 @@ import SwiftUI
...
-            ZStack {
-                Color.blue
+            VStack {
+                Text("👀 Brak donacji")
+                    .font(.largeTitle.bold())
+                Text("Trochę tu pusto, pora dodać jakąs donacje!")
+                    .font(.callout)
             }
-            .navigationTitle("Donations")
+            .navigationTitle("Lista donacji")
```
Comment (adjurk) [inline BloodDonor/Modules/DonationList/DonationListView.swift:19]: This should be changed to the agreed name of 'Dziennik donacji'

File: BloodDonor/Modules/DonationList/DonationListView.swift [hunk 2]
```diff
@@ -10,10 +10,13 @@ import SwiftUI
 struct DonationListView: View {
     var body: some View {
         NavigationView {
-            ZStack {
-                Color.blue
+            VStack {
+                Text("👀 Brak donacji")
+                    .font(.largeTitle.bold())
+                Text("Trochę tu pusto, pora dodać jakąs donacje!")
```
Comment (adjurk) [inline BloodDonor/Modules/DonationList/DonationListView.swift:16]: Typo in 'jakąś'

File: BloodDonor/Modules/Login/AuthHttpService.swift [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import Alamofire
+
+// Define a final class named AuthHttpService that conforms to the HttpService protocol.
+final class AuthHttpService: HttpService {
+    var sessionManager: Session = Session.default  // The session manager used for sending the request.
+    
+    // Function to send a request using the session manager and return a validated DataRequest.
+    func request(_ urlRequest: URLRequestConvertible) -> DataRequest {
+        return sessionManager.request(urlRequest).validate(statusCode: 200..<420)
```
Comment (adjurk) [inline BloodDonor/Modules/Login/AuthHttpService.swift:16]: The app should be able to handle `5xx` HTTP response codes in terms of server fuckups.

**Review by adjurk — DISMISSED** [review]
Couple of minor stuff (apart from the HTTP code handling) but overall an approve from me 👍

File: BloodDonor/Modules/DonationList/DonationListView.swift (hunk 1 above)
Comment (lukaszcettler) [reply BloodDonor/Modules/DonationList/DonationListView.swift:19]: oke will do

File: BloodDonor/Modules/DonationList/DonationListView.swift (hunk 2 above)
Comment (lukaszcettler) [reply BloodDonor/Modules/DonationList/DonationListView.swift:16]: mam dysleksje

File: BloodDonor/Modules/Login/AuthHttpService.swift (hunk 3 above)
Comment (lukaszcettler) [reply BloodDonor/Modules/Login/AuthHttpService.swift:16]: Ok, I will fix it in upcoming commits

**Review by adjurk — APPROVED** [review]

**Pull request merged**